src/String/number001/StringAPI.java

```diff
@@ -0,0 +1,127 @@
+package String.number001;
+
+import static String.number001.Constans.OUT_OF_BOUDS;
+
+/**
+ * Created By Arshak on Dec, 2020
+ */
+public class StringAPI {
+
+    //Number 1
+
+    String getCharAtIndex(String input, int k) {
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUDS;
+        }
+        if (k < 0 || k > input.length()) {
+            return OUT_OF_BOUDS;
+        }
+        return String.valueOf(input.charAt(k));
+    }
+
+    //Number 2
+    String wordsCount(String input) {
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUDS;
+        }
+
+        char[] chars = input.toCharArray();
+        int x = 1;
+        for (int i = 0; i < chars.length; i++) {
+            if (chars[i] == ' ') {
+                x = x + 1;
+
+            }
+
+        }
+        return String.valueOf(x);
+
+    }
+
+    //Number 3
+
+    String palindrome(String input) {
+        String lowerCase = input.toLowerCase();
+        StringBuilder builder = new StringBuilder(lowerCase);
+        builder.reverse();
+        String result = builder.toString();
+        if (result.equals(lowerCase)) {
+            return "Word is Palindrome";
+        } else return "Is not Palindrome";
+
+
+    }
+
+    //Number 4
+
+    String brackets(String input) {
+        int count = 0;
+        for (int i = 0; i < input.length(); i++) {
```

Comment:
sa el inchvor caser kpaki. + ches stugel null-n u empty-in

if(input.startsWith(")") || input.endsWith("(")) {
     return INVALID;
}